05week/bankAccount.js

```diff
@@ -0,0 +1,148 @@
+// BANK ACCOUNT CLASS
+
+class BankAccount {
+  constructor(acctNum, owner){
+    this.acctNum = acctNum 
+    this.owner = owner 
+    this.transactions = []
+  }
+
+  balance() { 
+    // returns the current balance on the account = sum of transactions array
+    return this.transactions
+    .reduce((prev,curr,i) => prev + curr, 0)
+  }
+
+  deposit(amt) {
+    // creates new transaction and adds to the transactions array.
+    // deposit must be greater than 0
+    if(amt > 0){
+      this.transactions.push(amt)
+    } else {
+      console.log('ALERT: Deposits must be a minimum of $1')
+    }
+  }
+
+  charge(amt, payee) {
+    //creates new transaction and adds to the transaction array
+    //cannot charge more than current balance
+    if(amt <= this.balance()){
+      this.transactions.push(-amt)
```

Comment:
this.transactions should hold the entire transaction object, not just the amount.